src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -592,28 +592,74 @@ bool Node::VerifyTxnsOrdering(const vector<TxnHash>& tranHashes) {
     }
   }
 
-  if (t_tranHashes == tranHashes) {
-    return true;
+  // check for txn misorder tolerance
+  const float TXN_MISORDER_TOLERANCE =
+      (float)TXN_MISORDER_TOLERANCE_IN_PERCENT / 100.00f;
+  const uint32_t MAX_MISORDER_TXNS =
+      std::ceil(t_tranHashes.size() * TXN_MISORDER_TOLERANCE);
+
+  LOG_GENERAL(INFO, "Tolerance = " << std::fixed << std::setprecision(2)
+                                   << TXN_MISORDER_TOLERANCE << " = "
+                                   << MAX_MISORDER_TXNS << " txns.");
+
+  uint32_t orderedtxns = 0;
+  uint32_t misordertxns = 0;
+  auto leftIt = t_tranHashes.begin();
+  auto rightIt = tranHashes.begin();
+  while (leftIt != t_tranHashes.end() && rightIt != tranHashes.end()) {
+    if (*leftIt == *rightIt) {
+      orderedtxns++;
+      leftIt++;
+      rightIt++;
+    } else {
+      break;
```

Comment:
I think @jiayaoqijia mentioned that this algorithm is not correct and that this is a Knapsack problem. Hence, we need to implement the algorithm to handle it

Reply:
@bb111189 We can apply this simple and strict algo first and later we further optimize it.